network.go

```diff
@@ -609,8 +612,31 @@ func (s *sandbox) updateRoute(netHandle *netlink.Handle, route *types.Route, add
 // DNS //
 /////////
 
-func setupDNS(dns []string) error {
-	return nil
+func setupDNS(dns []string) (err error) {
+	var file *os.File
+
+	if len(dns) == 0 {
+		agentLog.Debug("Did not set sandbox DNS as DNS not received as part of grpc request.")
+		return nil
+	}
+	if file, err = os.Create(kataGuestSandboxDNSFile); err != nil {
+		return err
+	}
+	defer file.Close()
+
+	for i, line := range dns {
+		if i != (len(dns) - 1) {
```

Comment:
nit: Its better to address the positive case first with reversing the order of the conditions.